### ISSUE TITLE
Introduce factory for Search Console response mock data

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2586]
After #2237, the `module-search-console-components.stories.js` file is now full of inlined fixtures, which is due to the mock data being dependent on the `referenceDate` used for the story. While we could have continued to use fixtures, doing so is in another way confusing as it decouples the two pieces entirely and obfuscates that they are connected.

We should introduce an Search Console-specific test utility, which generates Search Console response data.

We should also think about how this would affect request args definition in Storybook and tests - whether we should use fixtures there, factories, or just inline them (since those are _a lot_ simpler than response data).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be a function like `getSearchConsoleMockResponse( requestArgs, options )` in `tests/js/search-console.js` which generates a full Search Console response for the given request `requestArgs`. `options` would allow for additional tweaks to the response, if needed (TBD during IB / implementation).
    * This should be following a similar approach to #2584 and #2585, only for Search Console, which will require a detailed look at those responses and the format.
* The Search Console stories should use the above function instead of inlining response mock data.
* All Search Console fixtures for report data or args should be deleted from the plugin.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by danielgent** [comment]
None of this stories are used for VRTs so I'm going to add these first in my branch.  IMHO they should be merged to develop first as it will make the CR and QA for this branch way easier if we can just eyeball the :heavy_check_mark:s on GitHub

**Comment by danielgent** [comment]
I didn't realize how much this task was copying @eugene-manuilov's existing mock factories :smile: 

I'm happy to keep trying to copy his code, it looks really interesting but it uses libraries that I'm not familiar with

I could generate an array of data like this
```
{
    clicks: 3,
    ctr: 0,
    impressions: 6,
    keys: [
        '2020-07-01',
    ],
    position: 0,
},
```
fairly simply (I like Eugene's way of hashing the arguments to set the faker seed), but I think there's a lot of complexity in search console's response data that I'm missing.  As if it's like Analytics or AdSense then I'm going to have to use code like Eugene's

I'm happy to continue with this but I'd have to learn more about Google's data streams for search console, and the rxjs operations (assuming I copy Eugene's approach)

**Comment by eugene-manuilov** [comment]
> None of this stories are used for VRTs so I'm going to add these first in my branch. IMHO they should be merged to develop first as it will make the CR and QA for this branch way easier if we can just eyeball the s on GitHub

@danielgent there is no need to add VRTs for search console stories. This is out-of-scope for this ticket. Please, don't work on it. 

For this ticket, we need to focus only on the data mock factory for the search console module. It should generate reports similar to report data that we receive from real API. You can understand whether your data mock factory works correctly by checking [this](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/search-console-module-components-module-page-overview-widget--loaded) storybook story visually - if you see the graph that is similar to one from the `develop` branch, then you mock factory works correctly.

**Comment by eugene-manuilov** [comment]
> I didn't realize how much this task was copying @eugene-manuilov's existing mock factories 
> 
> I'm happy to keep trying to copy his code, it looks really interesting but it uses libraries that I'm not familiar with
> 
> I could generate an array of data like this
> 
> ```
> {
>     clicks: 3,
>     ctr: 0,
>     impressions: 6,
>     keys: [
>         '2020-07-01',
>     ],
>     position: 0,
> },
> ```
> 
> fairly simply (I like Eugene's way of hashing the arguments to set the faker seed), but I think there's a lot of complexity in search console's response data that I'm missing. As if it's like Analytics or AdSense then I'm going to have to use code like Eugene's
> 
> I'm happy to continue with this but I'd have to learn more about Google's data streams for search console, and the rxjs operations (assuming I copy Eugene's approach)

@danielgent I think we should stick with rxjs approach as we do for AdSense and Analytics modules because it allows us to easily generate data (metric values) for a sequence of dimension values (like a range of dates) and mix them together to form a set of rows for the report.

Check this youtube video to familiarize yourself with rxjs: https://youtu.be/ewcoEYS85Co. You will see that it is a very simple but in the same time, very powerful approach to generate data. Ping me on slack if you have questions or need help.

**Comment by aaemnnosttv** [comment]
Assigning this one to @eugene-manuilov since he's already done the other two 👍

**Comment by aaemnnosttv** [comment]
@eugene-manuilov I think this would be good to prioritize when you have a minute as I've seen this needed a few times recently but we're stuck using a fixture for now 😄 

Also, I've just updated the ACs to include the `provide*MockReport` helper as well like we have for the other modules.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv IB is added.

**Comment by asvinb** [comment]
### QA: ✅ 

* Search Console VRTs pass locally.
* Search Console Module stories and WP Dashboard stories render correctly.
